src/js/components/DataTable/buildState.js

```diff
@@ -84,12 +84,34 @@ export const filterAndSortData = (data, filters, onSearch, sort) => {
   if (sort && !sort.external) {
     const { property, direction } = sort;
     result = result === data ? [...data] : result; // don't sort caller's data
-    const before = direction === 'asc' ? 1 : -1;
-    const after = direction === 'asc' ? -1 : 1;
+    const sortAsc = direction === 'asc';
+    const before = sortAsc ? 1 : -1;
+    const after = sortAsc ? -1 : 1;
     result.sort((d1, d2) => {
-      if (datumValue(d1, property) > datumValue(d2, property)) return before;
-      if (datumValue(d1, property) < datumValue(d2, property)) return after;
-      return 0;
+      const d1Val = datumValue(d1, property);
+      const d2Val = datumValue(d2, property);
+      const d1IsString = typeof d1Val === 'string';
+      const d2IsString = typeof d2Val === 'string';
+      const d1ValProc = d1IsString ? d1Val.toLowerCase() : d1Val;
+      const d2ValProc = d2IsString ? d2Val.toLowerCase() : d2Val;
+      switch (true) {
```

Comment:
this is a really odd use of switch to do multiple bits of logic.  We should look at a cleaner way to do this. This is very hard to figure out compared to the original.

Reply:
Hey @MikeKingdom, the rule of thumb I was following is that when you have more than 3 cases in an if/then, a switch is generally considered preferable. That said, I see your point, and I realized there was a way to improve & simplify the logic a bit. I've pushed the change, so please review.